doc/specs/stdlib_hash_procedures.md

```diff
@@ -0,0 +1,1748 @@
+size of the structure in the hash code.
+This code is relatively fast on short keys, and is small enough that it
+will often be retained in the instruction cache if hashing is
+intermittent.
+As a result it should give good performance for typical hash table
+applications, although it is rare for them to need 64 bits.
+This code does not pass any of the SMHasher tests, but the resulting
+degradation in performance due to its larger number of collisions is
+expected to be minor compared to its faster hashing rate.
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+
+##### Example
+
+```fortran
+    program demo_fnv_1_hash
+      use stdlib_64_bit_hash_codes, only: fnv_1_hash
+	  use iso_fortran_env, only: int64
+      implicit none
+      integer, allocatable :: array1(:)
+      integer(int64) :: hash
+      array1 = [ 5, 4, 3, 1, 10, 4, 9]
+      hash = fnv_1_hash(array1)
+      print *, hash
+    end program demo_fnv_1_hash
+```
+
+
+#### `FNV_1A`- calculates a hash code from a key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 64 bit hash code from a rank 1 integer array or a default
+character string.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:fnv_1a]]( key )`
+
+##### Class
+
+Pure/elemental function
+
+##### Argument
+
+`key`: Shall be a deferred length default character scalar expression
+or a rank 1 integer array expression of kind `INT8`, `INT16`,
+`INT32`, or `INT64`.
+It is an `intent(in)` argument.
+
+##### Result
+
+The result is a scalar integer of kind `INT32`.
+
+##### Note
+
+`FNV_1A` is an implementation of the alternative FNV-1a hash code of
+Glenn Fowler, Landon Curt Noll, and Phong Vo.
+It differs from typical implementations in that it also encodes the
+size of the structure in the hash code.
+This code is relatively fast on short keys, and is small enough that it
+will often be retained in the instruction cache if hashing is
+intermittent.
+As a result it should give good performance for typical hash table
+applications.
+This code does not pass any of the SMHasher tests, but the resulting
+degradation in performance due to its larger number of collisions is
+expected to be minor compared to its faster hashing rate.
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+##### Example
+
+```fortran
+    program demo_fnv_1a_hash
+      use stdlib_64_bit_hash_codes, only: fnv_1a_hash
+      use iso_fortran_env, only: int64
+      implicit none
+      integer, allocatable :: array1(:)
+      integer(int64) :: hash
+      array1 = [ 5, 4, 3, 1, 10, 4, 9]
+      hash = fnv_1a_hash(array1)
+      print *, hash
+    end program demo_fnv_1a_hash
+```
+
+
+#### `NEW_PENGY_HASH_SEED`- returns a valid input seed for `PENGY_HASH`
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 32 bit "random" integer that is believed to be a valid
```

Comment:
I find the "believed to be" part unnecessary. I guess this is only the author reflecting upon the correctness of the code?

Reply:
The problem its that it is not always clear that there has been sufficient analysis/testing to be sure that there are no bad seeds.